pkg/middlewares/s3_bucket_acl_test.go

```diff
@@ -25,125 +25,174 @@ func TestS3BucketAcl_Execute(t *testing.T) {
 			name: "grant field on remote resource must be reset if acl != private in state resource",
 			args: args{
 				remoteResources: &[]resource.Resource{
-					&aws.AwsS3Bucket{
-						Id: "testgrant",
-						Grant: &[]struct {
-							Id          string   `cty:"id"`
-							Permissions []string `cty:"permissions"`
-							Type        *string  `cty:"type"`
-							Uri         *string  `cty:"uri"`
-						}{
-							{
-								"356616ba70ebbea29732c95eef24f9ea326b9018c167651705348b5af406a6db",
-								[]string{"FULL_CONTROL"},
-								awssdk.String("CanonicalUser"),
-								awssdk.String(""),
+					&resource.AbstractResource{
+						Id:   "testgrant",
+						Type: aws.AwsS3BucketResourceType,
+						Attrs: &resource.Attributes{
+							"grant": []map[string]interface{}{
+								{
+									"id":          "356616ba70ebbea29732c95eef24f9ea326b9018c167651705348b5af406a6db",
+									"permissions": []string{"FULL_CONTROL"},
+									"type":        awssdk.String("CanonicalUser"),
+									"uri":         awssdk.String(""),
+								},
```

Comment:
You can delete all the pointers for the attributes of each `grant` we won't need it anymore. Same for the other tests in this file.